mantidproject/mantid
# Create Basics of Fit Script Generator Interface

**Description of work.**
This PR does several things relating to the creation of a basic Fit Script Generator interface:
- Creates the initial layout of the Fit Script Generator interface.
- Exposes the interface to python so it can be opened from the Muon Analysis interface.
- Adds functionality to the data table on the left hand side of the interface (i.e. ability to remove and add a workspace domain).

![image](https://user-images.githubusercontent.com/40830825/100782325-29290600-3404-11eb-94cb-36e93fd231cd.png)

I also add tests for:
- `FitScriptGeneratorPresenter`
- `FitScriptGeneratorView`
- `FitScriptGeneratorDataTable`

Tests for `FitScriptGeneratorModel` have been omitted for now as this is likely to change a large amount in the future. 

**To test:**
1. Comment out Line 52 in `fitting_tab_view.py`
1. `Muon Interface`
2. Load any data and go to fitting tab.
3. Click `Fit Generator` to open the new interface
4. When testing this PR, only test the data table and buttons on the left hand side as this is the only thing that has been implemented so far. Things to test:
- Adding a workspace using the `Add Workspace` button
- Removing row from the table by selecting a row/multiple rows in the table, and then clicking `Remove`
- Try changing the `StartX` and `EndX` to values outside of the x range of a workspace. This should not be allowed.

The right hand side browsers will be connected up in a later PR in order to avoid a large PR.

*No release notes required as this is not a finished interface*

Fixes #29906

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
